web_monitoring/tests/test_html_diff_validity.py

```diff
@@ -175,3 +175,85 @@ def test_html_diff_works_on_documents_with_no_body():
+        '20171203125801/https://en.wikipedia.org/wiki/Mars_(mythology)" title='
+        '"Mars (mythology)">Mars (mythology)</a>. For other uses, see <a href='
+        '"/web/20171203125801/https://en.wikipedia.org/wiki/Mars_'
+        '(disambiguation)" class="mw-disambig" title="Mars (disambiguation)">'
+        'Mars (disambiguation)</a>.</div>',
+        include='all', strict_urls='WBM')
+
+    assert results['change_count'] == 0
+
+
+def test_html_diff_works_with_wbm_srcset():
+    results = html_diff_render(
+        '<img alt="OSIRIS Mars true color.jpg" srcset="//web-beta.archive.org/'
+        'web/20171105043925im_/https://upload.wikimedia.org/wikipedia/commons/'
+        'thumb/0/02/OSIRIS_Mars_true_color.jpg/413px-OSIRIS_Mars_true_color'
+        '.jpg 1.5x, //web-beta.archive.org/web/20171105043925im_/https://'
+        'upload.wikimedia.org/wikipedia/commons/thumb/0/02/OSIRIS_Mars_true_'
+        'color.jpg/550px-OSIRIS_Mars_true_color.jpg 2x" data-file-width="2205"'
+        ' data-file-height="2205" width="275" height="275">',
+        '<img alt="OSIRIS Mars true color.jpg" srcset="//web-beta.archive.org/'
+        'web/20171203125801im_/https://upload.wikimedia.org/wikipedia/commons/'
+        'thumb/0/02/OSIRIS_Mars_true_color.jpg/413px-OSIRIS_Mars_true_color'
+        '.jpg 1.5x, //web-beta.archive.org/web/20171203125801im_/https://'
+        'upload.wikimedia.org/wikipedia/commons/thumb/0/02/OSIRIS_Mars_true_'
+        'color.jpg/550px-OSIRIS_Mars_true_color.jpg 2x" data-file-width="2205"'
+        ' data-file-height="2205" width="275" height="275">',
+        include='all', strict_urls='WBM')
+
+    assert results['change_count'] == 0
+
+
+def test_html_diff_works_with_srcset():
+    results = html_diff_render(
+        '<img alt="OSIRIS Mars true color.jpg" src="https://upload.wikimedia.'
+        'org/wikipedia/commons/thumb/0/02/OSIRIS_Mars_true_color.jpg/413px-'
+        'OSIRIS_Mars_true_color.jpg">',
+        '<img alt="OSIRIS Mars true color.jpg" src="https://upload.wikimedia.'
+        'org/wikipedia/commons/thumb/0/02/OSIRIS_Mars_true_color.jpg/275px-'
+        'OSIRIS_Mars_true_color.jpg" srcset="https://upload.wikimedia.org/'
+        'wikipedia/commons/thumb/0/02/OSIRIS_Mars_true_color.jpg/413px-OSIRIS_'
+        'Mars_true_color.jpg 1.5x, https://upload.wikimedia.org/wikipedia/'
+        'commons/thumb/0/02/OSIRIS_Mars_true_color.jpg/550px-OSIRIS_Mars_true_'
+        'color.jpg 2x" data-file-width="2205" data-file-height="2205" width='
+        '"275" height="275">',
+        include='all')
+
+    assert results['change_count'] == 0
+
+
+def test_html_diff_works_with_jsessionid():
+    results = html_diff_render(
+        '<a href="https://www.ncdc.noaa.gov/homr/api;jsessionid=A2DECB66D2648B'
+        'FED11FC721FC3043A1"></a>',
+        '<a href="https://www.ncdc.noaa.gov/homr/api;jsessionid=45312D9542FDB0'
+        '15289A1BBD76958F43"></a>',
```

Comment:
Same formatting note as above. It’s OK to go over line length if it’s easier to read.